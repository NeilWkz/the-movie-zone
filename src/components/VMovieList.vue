<script>
import VMovieListItem from "@/components/VMovieListItem";
import _ from "lodash";

export default {
  name: "MovieList",
  props: {
    movies: {
      type: Array,
      required: true
    }
  },
  components: {
    VMovieListItem
  },
  computed: {
    orderedMovies: function() {
      let ordered = this.movies;
      ordered = _.orderBy(ordered, ["popularity"], ["desc"]);
      return ordered;
    }
  }
};
</script>
<template>
  <div class="movie-list">
    <VMovieListItem v-for="(movie) in orderedMovies" :movie="movie" :key="movie.id"/>
  </div>
</template>
