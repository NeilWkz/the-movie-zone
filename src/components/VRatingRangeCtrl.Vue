<script>

    export default {
        name: 'RatingRangeCtrl',
      
        props: {
            rating: {
                type: Object,
                required: true,
                default: () => ({})
            }
        },
          data: function () {
      return {
        value: 5,
      }
    }
    }
 
</script>

<template>
    <div>
        Rating Range Ctrl
        <vue-slider v-model="value"
        :min="0"
        :max="10"
        :interval="0.5"
        ></vue-slider>
    </div>
</template>