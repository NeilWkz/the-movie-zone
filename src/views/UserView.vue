<script>
import VMovieList from "@/components/VMovieList";
import VRatingRangeCtrl from "@/components/VRatingRangeCtrl";
export default {
  name: "UserView",
  components: {
    VMovieList,
    VRatingRangeCtrl
  },
  data() {
    return {
      rating: { initial: "3" }
    };
  }
};
</script>
<template>
  <div>
   
    <VRatingRangeCtrl :rating="rating"/>
     <VMovieList/>
  </div>
</template>