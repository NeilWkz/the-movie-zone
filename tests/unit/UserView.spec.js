import { shallowMount } from "@vue/test-utils";
import UserView from "@/views/UserView";
import VRatingRangeCtrl from "@/components/VRatingRangeCtrl";
import VMovieList from "@/components/VMovieList";
describe("UserView", () => {
  const build = () => {
    const wrapper = shallowMount(UserView);

    return {
      wrapper,
      userMovieList: () => wrapper.find(VMovieList),
      userRatingRangeCtrl: () => wrapper.find(VRatingRangeCtrl)
    };
  };

  it("renders the component", () => {
    // arrange
    const { wrapper } = build();
    // assert
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders main child components", () => {
    // arrange
    const { userMovieList, userRatingRangeCtrl } = build();

    // assert
    expect(userMovieList().exists()).toBe(true);
    expect(userRatingRangeCtrl().exists()).toBe(true);
  });
});
