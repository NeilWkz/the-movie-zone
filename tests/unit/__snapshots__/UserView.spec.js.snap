// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserView renders the component 1`] = `
<div>
  <hero-stub></hero-stub>
  <div class="wrap-controls container">
    <div id="nav-icon" role="button" aria-haspopup="true"><span></span> <span></span> <span></span> <span></span></div>
    <div aria-controls="movieList" class="controls">
      <div class="drawer-on-mobile container">
        <div class="filter-checkboxes">
          <h4 class="text-center">Filter By Genre</h4>
          <form id="genre-filters" aria-label="Genre Filters">
            <ul></ul>
          </form>
        </div>
        <div class="row justify-content-center">
          <div class="col-sm-6">
            <h4 class="text-center">Filter By Rating:</h4>
            <div class="rating-wrap">
              <vstarrating-stub rating="0"></vstarrating-stub>
            </div>
            <vue-slider-stub value="0" direction="ltr" dotsize="14" min="0" max="10" interval="0.5" clickable="true" duration="0.5" lazy="true" tooltip="focus" enablecross="true" order="true" process="true"></vue-slider-stub>
          </div>
        </div>
      </div>
    </div>
  </div>
  <!---->
  <div>
    <vnoresults-stub></vnoresults-stub>
  </div>
</div>
`;
